nemo-physical/src/tabular/operations/filter.rs

```diff
@@ -131,7 +131,11 @@ impl GeneratorFilter {
     /// Helper function that takes a list of boolean [Filter]s
     /// and constructs a [Filter] that represent its conjuction.
     fn fold_filters(filters: Vec<&Filter>) -> Filter {
-        Filter::boolean_conjunction(filters.into_iter().cloned().collect::<Vec<_>>())
+        if filters.len() == 1 {
+            filters[0].clone()
+        } else {
+            Filter::boolean_conjunction(filters.into_iter().cloned().collect::<Vec<_>>())
```

Comment:
Would it make sense to handle this special case within the `Filter::boolean_conjunction` function?

Reply:
I would like to keep the constructors simple. Doing such "optimizations" for conjunctions would probably also mean one would have to do them for all functions (for consistency :)). 

Also it should be fine for most cases to have a conjunction of one thing; it just breaks something in this specific instance, so I think it is appropriate to handle this here

Reply:
I have now included that check for all n-ary functions